chefkoch/recipe.py

```diff
@@ -0,0 +1,771 @@
+    values = [] #: All possible values of the parameter
+    
+    def __init__(self, name, entry):
+        """
+        Creates a new paramter from the JSON data gotten from the flavour file.
+        
+        :param name:        name as provided as in flavour file
+        :type name:         string
+        :param entry:       if the falvour file was a dict, it was flavour['name']
+        :type entry:        dictionary
+        """
+        self.values = []
+        logger.debug("Creating a new parameter " + str(name))
+        self.name = name
+        if type(entry) is not list: # a single value so to say
+            logger.debug("It has a single value: " + str(entry))
+            self.appendEntry(entry)
+        else:
+            logger.debug("It has more than one value.")
+            for sub_entry in entry: # the entry in the json file
+                self.appendEntry(sub_entry)
+        logger.debug("Fin.")
+
+    def appendFileParam(self, entry):
+        """
+        Appends a file parameter given in the JSON data to the Param.values
+        list.
+        
+        :param entry:    dict with fields type, file and key
+        :type entry:     dictionary
+        :raises ValueError: If there is no 'file' field to the entry or if the
+            given file path does not exist
+        """
+        try:
+            newValue = FileParamValue(entry['file'], entry['key'])
+            self.values.append(newValue)
+            logger.debug("Appending " + str(newValue) + newValue.tostring())
+        except KeyError as err:
+            # KeyError because the key-field is missing, in this case ignore
+            # todo: different possible exceptions
+            logger.exception("Either the file or the key field of the " +
+                             "entry are missing.")
+            # only abort if the file is missing. Missing keyword is possible.
+            try:
+                forgetfile = entry['file']
+            except KeyError as err:
+                raise ValueError("There is no file given to the file param!")
+                # todo: is it important to have a value error or can I change
+                # it to KeyError?
+            pass
+        except IOError as err:
+            # given filepath does not exist, so stop execution
+            raise ValueError("The filepath to the file param does not exist.")
+            # todo ValueError is catched somewhere. Rename to IOError there!
+
+    def appendValuesFromRange(self, entry):
+        """
+        Appends all values within a range given in the JSON data to Param.values
+        
+        :param entry:       dict with fields start, stop and step.
+        :type entry:        dictionary
+        :raises KeyError: If a field of the entry is missing.
+        """
+        logger.debug("More values are given by a range.")
+        try:
+            i = entry['start']
+            stop = entry['stop']
+            step = entry['step']
```

Comment:
you might want to consider defaults for the entries, i.e.
```
i, stop, step = entry.get('start', 0), entry.get('stop', -1), entry.get('step', 1)
```

Why is it named `i` and not `start`, which I would have expected to be more consistent and verbose?